packages/react/src/products/hooks/useProductDetails.ts

```diff
@@ -1,138 +1,86 @@
-    isProductFetched(state, id),
-  );
-  const isHydrated = useSelector((state: StoreState) =>
-    isProductHydrated(state, id),
-  );
-  const isInBag = useSelector((state: StoreState) => isProductInBag(state, id));
-  const isLoading = useSelector((state: StoreState) =>
-    isProductLoading(state, id),
+    isProductFetched(state, productId),
   );
-  const isOneSize = useSelector((state: StoreState) =>
-    isProductOneSize(state, id),
+  const product = useSelector((state: StoreState) =>
+    getProduct(state, productId),
   );
   const isOutOfStock = useSelector((state: StoreState) =>
-    isProductOutOfStock(state, id),
-  );
-  const labelsPrioritized = useSelector((state: StoreState) =>
-    getProductLabelsByPriority(state, id),
+    isProductOutOfStock(state, productId),
   );
-  const product = useSelector((state: StoreState) => getProduct(state, id));
-  const promotions = useSelector((state: StoreState) =>
-    getProductPromotions(state, id),
+  const isOneSize = useSelector((state: StoreState) =>
+    isProductOneSize(state, productId),
   );
-  const availableSizes = useSelector((state: StoreState) =>
-    getAllProductSizesRemainingQuantity(state, id),
+
+  const refetch = useCallback(
+    () => fetch(productId, fetchQuery, fetchForceDispatch, fetchConfig),
+    [fetch, productId, fetchQuery, fetchForceDispatch, fetchConfig],
   );
 
+  const shouldLoadDetails = enableAutoFetch && !isLoading && !error && !product;
+
   useEffect(() => {
-    !isFetched && fetchProductDetails(id);
-  }, [fetchProductDetails, id, isFetched]);
+    shouldLoadDetails &&
+      fetch(productId, fetchQuery, fetchForceDispatch, fetchConfig);
+  }, [
+    fetch,
+    productId,
+    fetchQuery,
+    fetchConfig,
+    shouldLoadDetails,
+    fetchForceDispatch,
+  ]);
 
   return {
-    /**
-     * The available sizes for the given product.
-     */
-    availableSizes,
-    /**
-     * Product breadcrumbs.
-     */
-    breadcrumbs,
-    /**
-     * Error state of the fetched product details.
-     */
     error,
-    /**
-     * Product grouped entries.
-     */
-    groupedEntries,
-    /**
-     * Whether the product is duplicated.
-     */
-    isDuplicated,
-    /**
-     * Whether the product has been fetched.
-     */
     isFetched,
-    /**
-     * Whether the product has been hydrated.
-     */
-    isHydrated,
-    /**
-     * Whether the product is in the bag.
-     */
-    isInBag,
-    /**
-     * Whether the product is loading.
-     */
     isLoading,
-    /**
-     * Whether the product has only one size.
-     */
-    isOneSize,
-    /**
-     * Whether the product is out of stock.
-     */
-    isOutOfStock,
-    /**
-     * Product labels on an ascended order.
-     */
-    labelsPrioritized,
-    /**
-     * The product itself.
-     */
-    product,
-    /**
-     * Product promotions.
-     */
-    promotions,
+    data: !!product
+      ? {
+          ...product,
+          isOneSize,
```

Comment:
What do you think of having the logic for `isOneSize` and `isOutOfStock` in the normalizer for the fetchProductDetailsAction so that the value will be available in the product entity, even when this hook is not used?

Reply:
IMO, putting these flags in the product entity is unnecessary. If someone doesn't want to use the hook and want to use the selectors instead, these two flags can be built with the information provided. 
I think that putting them in the store, is "replicating" information. What do you guys think?

Reply:
My idea was to remove those selectors altogether and have the normalizer extrapolate those values so the consumers would look at those flags directly on the product entity. I am still unsure if that normalization should be done at the client instead of doing it at the redux action. What do you think?

Reply:
As discussed offline, this idea might be a little tricky to implement because those properties are computed values so it would be best to not have them on the redux store although it is not very ergonomic to use the selectors to get those properties when we already have the product entity which contains many properties that do not need a selector to obtain. It would be much more interesting to have the Product entity be an instance with some get accessors which would calculate those derived properties but it would be tricky to have them in redux, so let's keep it the way it is now.